### ISSUE TITLE
Warnings about peer dependencies when using Cloudflare adapter in Astro 5.0.4

### ISSUE THREAD

**Issue by mikeavena** [withastro/adapters#477]
### Astro Info

```block
Astro                    v5.0.4
Node                     v21.7.3
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/partytown
                         @astrojs/sitemap
```

### Describe the Bug

After upgrading to Astro 5.0.4, npm now warns of peer dependencies related to the Cloudflare adapter. I had to force the upgrade to the latest version of Astro by doing `npm upgrade astro` then `npx @astrojs/upgrade`. Running just `npx @astrojs/upgrade` failed due to peer dependencies.

Full error message from `npm i`:

```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @inox-tools/astro-when@0.2.4
npm WARN Found: astro@5.0.4
npm WARN node_modules/astro
npm WARN   astro@"^5.0.4" from the root project
npm WARN   1 more (@astrojs/cloudflare)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer astro@"^4" from @inox-tools/astro-when@0.2.4
npm WARN node_modules/@inox-tools/astro-when
npm WARN   @inox-tools/astro-when@"^0.2.4" from @astrojs/cloudflare@12.0.1
npm WARN   node_modules/@astrojs/cloudflare
npm WARN 
npm WARN Conflicting peer dependency: astro@4.16.17
npm WARN node_modules/astro
npm WARN   peer astro@"^4" from @inox-tools/astro-when@0.2.4
npm WARN   node_modules/@inox-tools/astro-when
npm WARN     @inox-tools/astro-when@"^0.2.4" from @astrojs/cloudflare@12.0.1
npm WARN     node_modules/@astrojs/cloudflare
npm WARN ERESOLVE overriding peer dependency
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@astrojs/prism@3.2.0',
npm WARN EBADENGINE   required: { node: '^18.17.1 || ^20.3.0 || >=22.0.0' },
npm WARN EBADENGINE   current: { node: 'v21.7.3', npm: '10.5.0' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@astrojs/telemetry@3.2.0',
npm WARN EBADENGINE   required: { node: '^18.17.1 || ^20.3.0 || >=22.0.0' },
npm WARN EBADENGINE   current: { node: 'v21.7.3', npm: '10.5.0' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'astro@5.0.4',
npm WARN EBADENGINE   required: {
npm WARN EBADENGINE     node: '^18.17.1 || ^20.3.0 || >=22.0.0',
npm WARN EBADENGINE     npm: '>=9.6.5',
npm WARN EBADENGINE     pnpm: '>=7.1.0'
npm WARN EBADENGINE   },
npm WARN EBADENGINE   current: { node: 'v21.7.3', npm: '10.5.0' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'vite@6.0.3',
npm WARN EBADENGINE   required: { node: '^18.0.0 || ^20.0.0 || >=22.0.0' },
npm WARN EBADENGINE   current: { node: 'v21.7.3', npm: '10.5.0' }
npm WARN EBADENGINE }
```

### What's the expected result?

Expected no warnings or errors related to dependencies.

### Link to Minimal Reproducible Example

N/A

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by alexanderniebuhr** [comment]
@Fryuni Haven't you updated inox tools yet. We are on the latest version as far as I can tell 🤔

**Comment by Fryuni** [comment]
Astro When v0.2.4 is the latest version compatible with Astro 4.
v1.0.1 is the latest compatible with Astro 5.

**Comment by Fryuni** [comment]
AFAICT Cloudflare adapter is still not ready for Astro 5 right? That is the reason for https://github.com/withastro/adapters/pull/454. I mentioned in the PR that it would require updating Astro When as well.

**Comment by alexanderniebuhr** [comment]
The Cloudflare adapter is ready for v5. And we have updated to `v1.0.1` _(see main branch, but people still get the warning about the peer dependencies?)_

**Comment by Fryuni** [comment]
Oh, it's updated on main but there is no release since

**Comment by alexanderniebuhr** [comment]
Oh my bad, we should release then 😅

**Comment by mikeavena** [comment]
This appears to be fixed with version 12.1.0
